src/standalone/Vegetation/pmodel.jl

```diff
@@ -337,4 +431,568 @@ function compute_full_pmodel_outputs(
+    T_canopy = canopy_temperature(canopy.energy, canopy, Y0, p)
+    VPD = @. lazy(
+        ClimaLand.vapor_pressure_deficit(
+            p.drivers.T,
+            p.drivers.P,
+            p.drivers.q,
+            LP.thermodynamic_parameters(canopy.parameters.earth_param_set),
+        ),
+    )
+    I_abs = @. lazy(
+        compute_I_abs(
+            p.canopy.radiative_transfer.par.abs,
+            p.canopy.radiative_transfer.par_d,
+            canopy.radiative_transfer.parameters.λ_γ_PAR,
+            constants.lightspeed,
+            constants.planck_h,
+            constants.N_a,
+        ),
+    )
+
+    # Initialize OptVars with dummy values (will be overwritten since α=0)
+    fill!(
+        p.canopy.photosynthesis.OptVars,
+        (; ξ_opt = FT(0), Vcmax25_opt = FT(0), Jmax25_opt = FT(0)),
+    )
+
+    parameters_init = PModelParameters(
+        cstar = parameters.cstar,
+        β = parameters.β,
+        ϕc = parameters.ϕc,
+        ϕ0 = parameters.ϕ0,
+        ϕa0 = parameters.ϕa0,
+        ϕa1 = parameters.ϕa1,
+        ϕa2 = parameters.ϕa2,
+        α = FT(0),  # this allows us to use the initial values directly
+        sc = parameters.sc,
+        pc = parameters.pc,
+    )
+
+    local_noon_mask = FT(1)  # Force update for initialization
+
+    @. p.canopy.photosynthesis.OptVars = update_optimal_EMA(
+        parameters_init,
+        constants,
+        p.canopy.photosynthesis.OptVars,
+        T_canopy,
+        p.drivers.P,
+        VPD,
+        p.drivers.c_co2,
+        βm,
+        I_abs,
+        local_noon_mask,
+    )
+end
+
+
+function call_update_optimal_EMA(
+    p,
+    Y,
+    t;
+    canopy,
+    dt,
+    local_noon,
+    verbose = false,
+)
+    local_noon_mask = @. lazy(get_local_noon_mask(t, dt, local_noon))
+
+    # update the acclimated Vcmax25, Jmax25, ξ using EMA 
+    parameters = canopy.photosynthesis.parameters
+    constants = canopy.photosynthesis.constants
+
+    # drivers 
+    FT = eltype(parameters)
+    # TODO: replace this with modular soil moisture stress parameterization
+    ψ = p.canopy.hydraulics.ψ
+    n = canopy.hydraulics.n_leaf + canopy.hydraulics.n_stem
+    grav = LP.grav(earth_param_set)
+    ρ_water = LP.ρ_cloud_liq(earth_param_set)
+    βm = @. lazy(moisture_stress(ψ.:($$n) * ρ_water * grav, parameters.sc, parameters.pc))
+    T_canopy = canopy_temperature(canopy.energy, canopy, Y, p)
+    VPD = @. lazy(
+        ClimaLand.vapor_pressure_deficit(
+            p.drivers.T,
+            p.drivers.P,
+            p.drivers.q,
+            LP.thermodynamic_parameters(canopy.parameters.earth_param_set),
+        ),
+    )
+    I_abs = @. lazy(
+        compute_I_abs(
+            p.canopy.radiative_transfer.par.abs,
+            p.canopy.radiative_transfer.par_d,
+            canopy.radiative_transfer.parameters.λ_γ_PAR,
+            constants.lightspeed,
+            constants.planck_h,
+            constants.N_a,
+        ),
+    )
```

Comment:
It looks like all of this is copied in `set_historical_cache!` too so it could be pulled out into a function they both call to reduce code duplication. But since it's only in 2 places it's ok like this too